docs/architecture/decisions/0014-dynamic-datasource-registration.md

```diff
@@ -0,0 +1,106 @@
+# 0014. Dynamic Data Source Registration
+
+## Status
+
+Accepted
+
+## Context
+
+Decouple the data sources from the SDK into their own NuGet packages and create a mechanism to allow auto discovery and dynamic registration. This will allow for new data sources to be plugged in with minimum configuration and dependency on the existing code base. 
```

Comment:
Since auto discovery and dynamic registration are already part of the approved [ADR 0006](https://github.com/Green-Software-Foundation/carbon-aware-sdk/blob/dev/docs/architecture/decisions/0006-data-source-registration.md) I am wondering if the intent of this ADR should be more focused on the additional bit of moving the current data sources out. What do you think, @juzuluag ?

Reply:
> Since auto discovery and dynamic registration are already part of the approved [ADR 0006](https://github.com/Green-Software-Foundation/carbon-aware-sdk/blob/dev/docs/architecture/decisions/0006-data-source-registration.md) I am wondering if the intent of this ADR should be more focused on the additional bit of moving the current data sources out. What do you think, @juzuluag ?

Hey @bderusha , sorry for the late response. I'll get back to you soon with answers to your questions.

Reply:
Hi @bderusha, agree. I changed the context section